presto-main/src/main/java/io/prestosql/sql/planner/sanity/DynamicFiltersChecker.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner.sanity;
+
+import com.google.common.collect.ImmutableSet;
+import io.prestosql.Session;
+import io.prestosql.execution.warnings.WarningCollector;
+import io.prestosql.metadata.Metadata;
+import io.prestosql.sql.DynamicFilters;
+import io.prestosql.sql.planner.TypeAnalyzer;
+import io.prestosql.sql.planner.TypeProvider;
+import io.prestosql.sql.planner.plan.FilterNode;
+import io.prestosql.sql.planner.plan.JoinNode;
+import io.prestosql.sql.planner.plan.PlanNode;
+import io.prestosql.sql.planner.plan.PlanVisitor;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.Sets.difference;
+import static com.google.common.collect.Sets.intersection;
+import static io.prestosql.sql.DynamicFilters.extractDynamicFilters;
+
+/**
+ * When dynamic filter assignments are present on a Join node, they should be consumed by a Filter node on it's probe side
+ */
+public class DynamicFiltersChecker
+        implements PlanSanityChecker.Checker
+{
+    @Override
+    public void validate(PlanNode plan, Session session, Metadata metadata, TypeAnalyzer typeAnalyzer, TypeProvider types, WarningCollector warningCollector)
+    {
+        plan.accept(new PlanVisitor<Set<String>, ImmutableSet<String>>()
+        {
+            @Override
+            protected Set<String> visitPlan(PlanNode node, ImmutableSet<String> expected)
+            {
+                Set<String> consumed = new HashSet<>();
+                for (PlanNode source : node.getSources()) {
+                    consumed.addAll(source.accept(this, expected));
+                }
+                return consumed;
+            }
+
+            @Override
+            public Set<String> visitJoin(JoinNode node, ImmutableSet<String> expected)
+            {
+                ImmutableSet<String> currentJoinDynamicFilters = ImmutableSet.copyOf(node.getDynamicFilters().keySet());
+                ImmutableSet<String> expectedProbeSide = ImmutableSet.<String>builder()
```

Comment:
I think you just want to make sure that probe side consumed all of the current `JoinNode` dynamic filters.